polars/polars-lazy/polars-plan/src/dsl/function_expr/schema.rs

```diff
@@ -125,6 +125,8 @@ impl FunctionExpr {
                     Month | Quarter | Week | WeekDay | Day | OrdinalDay | Hour | Minute
                     | Millisecond | Microsecond | Nanosecond | Second => DataType::UInt32,
                     TimeStamp(_) => DataType::Int64,
+                    Truncate(..) => DataType::Date,
```

Comment:
Ok I think we are close now (fingers crossed) however I suspect this is incorrect as it can return either a date or datetime. I tried using same_dtype but that was returning field rather than a dtype. What is the correct method required here?

Reply:
Ok I have unwrapped and gotten the dtype from same_type. Is this the correct implementation?

Reply:
@ritchie46

Reply:
This should be `same_type`.

Reply:
It should be in the most up to date one